client/components/user/signup/signup.component.js

```diff
@@ -31,28 +28,39 @@ angular.module('user').component('signup', {
           }
         }
       };
-
-      this.getError = function(error) {
+         this.getErroremail = function(error) {
         if (angular.isDefined(error)) {
           if (error.required) {
             return 'Please fill up this field';
           } else if (error.email) {
             return 'Please input correct email';
+          } else if (error.pattern) {
+            return 'Only top level domein is allowed';
+          } 
+        }
+      };
+
+      this.getError = function(error) {
```

Comment:
not a very good architecture. You should have kind a object with messages and type sand get message by type from that constant.